### PR TITLE
Some more work in the geometries

### PR DESCRIPTION
**Pull request by joto**
See individual commits.

### CODE REVIEW

File: src/geom-from-osm.cpp [hunk 1]
```diff
@@ -96,27 +96,37 @@ geometry_t create_polygon(osmium::Way const &way)
...
+        }
+    } else {
+        auto &multiline = geom->set<multilinestring_t>();
+        for (auto const &way : ways) {
+            linestring_t line;
+            fill_point_list(&line, way.nodes());
+            if (line.size() >= 2U) {
+                multiline.add_geometry(std::move(line));
+            }
+        }
```
Comment (lonvia) [inline src/geom-from-osm.cpp:118]: There is a corner case here when there are multiple ways in the buffer but only one of them is valid. Then still a multilinestring with a single member is created. It's probably okay, just something we should keep in mind and potentially document.

File: tests/test-geom-multilinestrings.cpp [hunk 2]
```diff
@@ -36,6 +36,25 @@ TEST_CASE("create_multilinestring with single line", "[NoDB]")
     REQUIRE(ml[0] == expected);
 }
 
+TEST_CASE("create_multilinestring with single line an no force_multi", "[NoDB]")
```
Comment (lonvia) [inline tests/test-geom-multilinestrings.cpp:39]: an -> and

Comment (joto) [reply tests/test-geom-multilinestrings.cpp]: Fixed in new version.

File: src/geom-from-osm.cpp (hunk 1 above)
Comment (joto) [reply src/geom-from-osm.cpp:118]: I put it on my todo list.

**Pull request merged**